File: split/appendkern.pas
procedure appendkern;
var
  s: quarterword;
begin
  s := curchr;
  scandimen(s = 99, false, false);
  begin
    mem[curlist.tailfield].hh.rh := newkern(curval);
    curlist.tailfield := mem[curlist.tailfield].hh.rh;
  end;
  mem[curlist.tailfield].hh.b1 := s;
end;

File: cpp/appendkern.cpp
#include "appendkern.h"
#include "scandimen.h"
#include "newkern.h"

void appendkern(void)
{
	auto s = curchr;
	scandimen(s == 99, false, false);
	link(curlist.tailfield) = newkern(curval);
	curlist.tailfield = link(curlist.tailfield);
	subtype(curlist.tailfield) = s;
}
